api/models/site.js

```diff
@@ -42,6 +68,16 @@ function toJSON() {
   object.createdAt = object.createdAt.toISOString();
   object.updatedAt = object.updatedAt.toISOString();
 
+  // prevent any invalid domain or demoDomain URLs from being returned
+  // or used in the creation of viewLink or demoViewLink
+  if (!isEmptyOrUrl(object.domain, false)) {
+    object.domain = null;
+  }
+
+  if (!isEmptyOrUrl(object.demoDomain, false)) {
```

Comment:
These checks prevent any *currently* existing Site models from returning invalid URLs for `domain` and `demoDomain` (since the new `validate` config below will only happen on model creation or update).

@jmhooper checked our production and staging databases and found no maliciously-invalid `domain` or `demoDomain` fields, however.